### PR TITLE
Upgrade GDP Master to yocto 2.2 (morty)

### PR DESCRIPTION
**Pull request by chbae**
[GDP-550] Upgrade GDP Master to yocto 2.2 (morty)

### CODE REVIEW

**Comment by chbae** [comment]
This PR should be tested with https://github.com/GENIVI/genivi-dev-platform/pull/67

I tested on qemux86-64, raspberrypi3, and minnowboard turbot but dragonboard is not working properly.
However I thought that those are enough for first review.

@gunnarx, @zeenix, @leon-anavi 
Could you review my changes? Thanks

**Comment by chbae** [comment]
I fixed dragonboard problem. Now PR is working on all devices (qemux86, raspberrypi3, minnowboard, dragonboard). We need to check renesas board.

@slawr Could you write M3 build guide in GDP wiki?

**Review by zeenix — CHANGES_REQUESTED** [review]
The author didn't intend this patch to go as is. :) This is at least 3 separate changes so we should divide into 3 patches.

File: meta-genivi-dev/recipes-devtools/python/python-pyqt_5.3.1.bb
Comment (zeenix) [inline meta-genivi-dev/recipes-devtools/python/python-pyqt_5.3.1.bb]: Fixing formatting is always a welcomed change but they shouldn't be hidden in unrelated patches.

**Review by zeenix — CHANGES_REQUESTED** [review]
"So systemd is downgraded to v229 that is using in current master branch."

Sorry this part isn't very clear. What is using systemd 229 in master branch?

**Comment by zeenix** [comment]
Looks good in general. I'm wondering how we'll merge this work? g-d-p first or m-g-d first? For most similar PRs, I've preferred g-d-p first so if there is no strong feelings, I'll just stick to that to be consistent.

**Comment by zeenix** [comment]
> Looks good in general. I'm wondering how we'll merge this work? g-d-p first or m-g-d first? For most similar PRs, I've preferred g-d-p first so if there is no strong feelings, I'll just stick to that to be consistent.

Actually in this case, it's probably best to merge m-g-d (this PR) first cause otherwise we'll have a broken build.

**Comment by chbae** [comment]
@zeenix If you agree to merge this PR, I'll merge m-g-d first and then merge g-d-p. I'll do it because g-d-p commit will be changed after merging m-g-d.

**Comment by chbae** [comment]
> this part isn't very clear. What is using systemd 229 in master branch?

current GDP (yocto jethro) is using systemd 229.

**Comment by chbae** [comment]
> The author didn't intend this patch to go as is. :) This is at least 3 separate changes so we should divide into 3 patches.

Sorry. I can't understand 3 separate changes. Do you mean 3 separated PRs?

**Comment by zeenix** [comment]
> @zeenix If you agree to merge this PR, I'll merge m-g-d first and then merge g-d-p. I'll do it because g-d-p commit will be changed after merging m-g-d.

Sure, isn't that what I suggested in my last comment (just before yours)?

> current GDP (yocto jethro) is using systemd 229.

Ah ok, let's modify the commit log slightly to make that clear then.

> Sorry. I can't understand 3 separate changes. Do you mean 3 separated PRs?

I meant 3 patches. This is re the first patch that has my name on it. :) Stupid github doesn't let me put general comment on the specific patches. :(

**Comment by chbae** [comment]
> I meant 3 patches. This is re the first patch that has my name on it. :) Stupid github doesn't let me put general comment on the specific patches. :(

The patch is to fix python-pyqt build error in morty and I thought that it's better to use just one commit. Nevertheless if you would like to separate to 3 commits, I'll separate it.

> Sure, isn't that what I suggested in my last comment (just before yours)?

Correct. Then I'll do it after your agreement about this PR.

> Ah ok, let's modify the commit log slightly to make that clear then.

Ok. I'll modify the log.

Thanks.

**Review by gunnarx — COMMENTED** [review]
This commit concerns me the most. 

https://github.com/GENIVI/meta-genivi-dev/pull/74/commits/5a3d9dd0e3f1532387246cdb88b6e1b16a9f6d3e

I wonder what we could do to avoid carrying such significant and many patches?

Many points involved here : The qcom BSP, weston, versions in poky and oe for morty, and maybe something GDP unique?

Is this only a downgrade of systemd or something more? Is there no layer that carries, or can be made to carry, the old version if it's only downgrade, I mean why does it have to be more than a _preferred_version = 229 type change I guess I'm wondering?

**Comment by gunnarx** [comment]
A personal reflection here - I feel like we are not yet at the stage of "improve commit messages".  How much design and impact discussion have you had on this yet, and should we look at that first (so as to not avoid additional rework if commit content needs to change because of it).

**Comment by chbae** [comment]
> Is this only a downgrade of systemd or something more? Is there no layer that carries, or can be made to carry, the old version if it's only downgrade, I mean why does it have to be more than a _preferred_version = 229 type change I guess I'm wondering?
I tried to downgrade only systemd but others (systemd-xxx) are also required. So I created layer. :(

**Comment by zeenix** [comment]
> The patch is to fix python-pyqt build error in morty and I thought that it's better to use just one commit. Nevertheless if you would like to separate to 3 commits, I'll separate it.

The philosophy is every change going into separate patch. There can be 1-1, 1-N or N-1 relationship between patches and issues/bugs/features. One of the main reasons of this philosophy is that if we want to revert changes, it's easier if those changes were isolated and usually when you revert, it's not to revert the intended fix/benefits of the patch but rather some undesirable side-effect.

**Comment by zeenix** [comment]
> A personal reflection here - I feel like we are not yet at the stage of "improve commit messages". How much design and impact discussion have you had on this yet, and should we look at that first (so as to not avoid additional rework if commit content needs to change because of it).

Knowing @chbae and the fact that he's reported so, I'm confident that he has tested these changes on his end so I'm not expecting complete re-write of all changes. Moreover, fixing commit messages also gives you a good exercise so common mistakes are avoided in future.

**Comment by zeenix** [comment]
> This commit concerns me the most. 5a3d9dd

:+1:

**Comment by gunnarx** [comment]
Knowing @chbae and the fact that he's reported so, I'm confident that he has tested these changes on his end so I'm not expecting complete re-write of all changes. Moreover, fixing commit messages also gives you a good exercise so common mistakes are avoided in future.

**Comment by chbae** [comment]
rebased
@zeenix, @gunnarx 
Please review those chagnes. Thanks.

**Comment by zeenix** [comment]
@chbae Thanks. Looks good on a quick look. I think the main issue we need to discuss is downgrade of systemd for dragonboard as @gunnarx also said. Could this not be handled better?

**Comment by slawr** [comment]
I see the systemd downgrade for dragonboard 47789dcdbf is related to Weston. I'm not familiar with the board code but two things come to mind. First is that Poky Weston changed it's method of starting Weston, which may be a factor if your updating manually. Second is that GDP does not handle Weston init properly (see https://at.projects.genivi.org/jira/browse/GDP-530) in its recipes and a change upstream may be breaking that weak link. Just some 2c guesses..

**Comment by chbae** [comment]
@zeenix I need to change not to use meta-qcom-gdp

**Comment by gunnarx** [comment]
FYI I also pushed this into a temporary [morty branch](https://github.com/GENIVI/meta-genivi-dev/tree/morty) for easier tracking and to ensure the g-d-p submodule has access to the needed commits when building.

**Comment by chbae** [comment]
@gunnarx @zeenix 
When you are ready to merge this change, I'll rebase and merge it.
I'm waiting for your review.

**Comment by chbae** [comment]
@zeenix @gunnarx 
Please give me your test and review. I'll apply it and then I'll start upgrading qt to v5.6.3.

**Comment by gunnarx** [comment]
Same as before, c850555c, (this PR rebased on latest master, and the same for meta-genivi-dev PR), builds OK for R-Pi, Dragonboard.  Intel minnowboard builds fine clean but not with reused sstate.  (Have a pretty good idea what needs to change in the conf to fix that though).

**Comment by gunnarx** [comment]
Build same as before, see https://github.com/GENIVI/genivi-dev-platform/pull/67#issuecomment-295133728

**Review by zeenix — APPROVED** [review]

**Pull request merged**
